apps/visualization/schema.py

```diff
@@ -81,6 +84,50 @@ async def get_country_profile_object(iso3):
         return None
 
 
+@sync_to_async
+def get_sources(iso3, emergency, indicator_id, indicator_name, indicator_description, subvariable, thematic, topic, type):
+    filters = clean_filters({
+        'iso3': iso3,
+        'emergency': emergency,
+        'indicator_id': indicator_id,
+        'indicator_name': indicator_name,
+        'indicator_description': indicator_description,
+        'subvariable': subvariable,
+        'themmatic': thematic,
+        'topic': topic,
+        'type': type,
+    })
+    data = DataGranular.objects.filter(
+        **filters
+    ).values(
+        'iso3',
+        'emergency',
+        'indicator_id',
+        'subvariable',
+        'thematic',
+        'topic',
+        'type'
+    ).annotate(latest_month=Max('indicator_month'))
+    if data:
+        filters = reduce(
+            lambda acc,
+            item: acc | item,
+            [
+                Q(
+                    iso3=value['iso3'],
+                    emergency=value['emergency'],
+                    indicator_id=value['indicator_id'],
+                    indicator_month=value['latest_month'],
+                    subvariable=value['subvariable'],
+                    thematic=value['thematic'],
+                    topic=value['topic'],
+                    type=value['type'],
+                ) for value in data
+            ]
+        )
+        return list(DataGranular.objects.filter(filters))
```

Comment:
Can you explain what is happening here? cc: @bimal125

Reply:
@thenav56 , Actually, we need a api to list the sources [DataGranular]. These objects should be the latest month data

Reply:
New change invalidates this review comment.